File: src/p2/xlport/shims/besselj0.cpp/shim.cpp
// Study-5 Family-XL adapter shim (pair: besselj0.cpp). Adapter/oracle layer only:
// calls the unmodified external entry point(s); frozen aux documented in
// docs/prereg_v2/STUDY5_XL_ROSTER.md.
#include <cstdio>
#include <cmath>
#include <boost/math/special_functions/bessel.hpp>

static double program(double x) {
    return boost::math::cyl_bessel_j(0, 8.0 * x - 4.0);
}

int main() {
    char buf[256];
    double x = 0.0, y = 0.0;
    while (std::fgets(buf, sizeof buf, stdin)) {
        if (std::sscanf(buf, "%lf", &x) != 1) continue;
        try { y = program(x); } catch (...) { y = std::nan(""); }
        std::printf("%.17g\n", y);
        std::fflush(stdout);
    }
    return 0;
}

File: src/p2/xlport/shims/quantile.jl/shim.jl
# Study-5 Family-XL adapter shim (pair: quantile.jl). Adapter/oracle layer only:
# calls the unmodified Julia stdlib entry point; frozen aux documented in
# docs/prereg_v2/STUDY5_XL_ROSTER.md.
using Statistics
using Printf

const V = [-3.7, -2.9, -2.3, -1.7, -1.3, -0.9, -0.6, -0.35, -0.2, -0.08, 0.0, 0.08, 0.2, 0.35, 0.6, 0.9, 1.3, 1.7, 2.3, 2.9, 3.7]

program(x) = quantile(V, x)

while !eof(stdin)
    line = readline(stdin)
    s = strip(line)
    isempty(s) && continue
    x = try
        parse(Float64, s)
    catch
        continue
    end
    y = program(x)
    @printf("%.17g\n", y)
    flush(stdout)
end
